Rename some methods in the WP_Theme_JSON_Resolver class

anton-vlasenko opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: https://core.trac.wordpress.org/ticket/54517

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

anton-vlasenko committed:
Fix names. Please see https://core.trac.wordpress.org/ticket/54517

hellofromtonya commented:
@anton-vlasenko can you resolve the merge conflicts please?

hellofromtonya commented:
@noisysocks what do you think of the new naming?

noisysocks commented:
> @noisysocks what do you think of the new naming?

We do typically avoid referencing CPTs in other parts of Core as it's usually more of an implementation detail.

I defer to @ockham who ported most of this 😀

One consideration is that we'd have to update Gutenberg with these changes too.

hellofromtonya commented:
>I defer to @ockham who ported most of this 😀

Hey @ockham, what do you think of renaming? 

>One consideration is that we'd have to update Gutenberg with these changes too.

@anton-vlasenko is there a matching PR in Gutenberg? Also can you fix the merge conflicts please?

ockham commented:
> > I defer to @ockham who ported most of this 😀
> 
> Hey @ockham, what do you think of renaming?

Hey, sorry for the late reply 😅 TBH I'm less familiar with the `theme.json` resolver -- that would be more @oandregal's or @jorgefilipecosta's area of expertise, so maybe they can weigh in.

(FWIW, personally I think that something like `get_something_from_post` makes sense, when e.g. compared to `get_user_something_from_custom_post_type`. The `_from_post` suffix is also something we do in template resolution, at least in one private function, `_build_block_template_result_from_post` (which has a counterpart, `_build_block_template_result_from_file`).)

Commit:
Merge branch 'trunk' into fix/naming-of-methods-in-wp-theme-json-resolver-is-not-optimal

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -263,13 +263,13 @@ public static function get_user_data_from_custom_post_type( $theme, $should_crea
 
 		// Special case: '-1' is a results not found.
 		if ( -1 === $post_id && ! $should_create_cpt ) {
```
```suggestion
		if ( -1 === $post_id && ! $create_post ) {
```
I missed this in resolving the merge conflict.

hellofromtonya commented on src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php:
```diff
@@ -318,7 +318,7 @@ public function prepare_item_for_response( $item, $request ) {
-			$id = WP_Theme_JSON_Resolver::get_user_custom_post_type_id();
+			$id = WP_Theme_JSON_Resolver::get_global_styles_post_id();
 		} else {
 			$user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_custom_post_type( $theme );
```
```suggestion
			$user_cpt = WP_Theme_JSON_Resolver::get_global_styles_from_post( $theme );
```
Missed this one when resolving the merge conflicts.

hellofromtonya committed:
Additional items from fixing merge conflicts.

hellofromtonya committed:
Update WP_Theme_JSON_Resolver::get_global_styles_from_post() tests.

hellofromtonya reviewed (approved):
The new method names are more descriptive of what each does without using "cpt" or "custom post type" terminology. I think this looks good.

hellofromtonya commented:
@anton-vlasenko These changes will need to be backported to Gutenberg. Can you take care of that?

oandregal commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -216,28 +216,28 @@ public static function get_theme_data( $deprecated = array() ) {
 	 *                                     so it only fetches published posts.
 	 * @return array Custom Post Type for the user's origin config.
 	 */
-	public static function get_user_data_from_custom_post_type( $theme, $should_create_cpt = false, $post_status_filter = array( 'publish' ) ) {
```
Hey, the proposed naming `get_global_styles_from_post` doesn't reflect what this method does. If we want to remove the "cpt" reference I'd suggest using `get_user_data_from_{name of the cpt method}`.

hellofromtonya replied:
@oandregal What about renaming to `get_user_data_from_wp_global_styles`, i.e. where `wp_global_styles` is the custom type post?

oandregal replied:
:+1:

oandregal commented on src/wp-admin/site-editor.php:
```diff
@@ -91,7 +91,7 @@ static function( $classes ) {
 	// Used in the HTML title tag.
 	$title = __( 'Editor (beta)' );
 
-	$active_global_styles_id = WP_Theme_JSON_Resolver::get_user_custom_post_type_id();
```
I'm not very fond of the new name proposed as it loses meaning. If we want to remove the "cpt" reference, what about `get_user_global_styles_post_id`? Or something along those lines. As long as it's clear that the data stored is from the user I'm fine.

hellofromtonya replied:
Done in commit https://github.com/WordPress/wordpress-develop/pull/1957/commits/2180b6eb005ea73f03dfd24c1fd336ab23e5b687

hellofromtonya commented on src/wp-admin/site-editor.php:
```diff
@@ -91,7 +91,7 @@ static function( $classes ) {
 	$title = __( 'Editor (beta)' );
 
-	$active_global_styles_id = WP_Theme_JSON_Resolver::get_user_custom_post_type_id();
+	$active_global_styles_id = WP_Theme_JSON_Resolver::get_global_styles_post_id();
```
```suggestion
	$active_global_styles_id = WP_Theme_JSON_Resolver::get_user_global_styles_post_id();
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -381,12 +381,12 @@ public static function get_merged_data( $origin = 'custom' ) {
 	 * @return integer|null
 	 */
-	public static function get_user_custom_post_type_id() {
+	public static function get_global_styles_post_id() {
```
```suggestion
	public static function get_user_global_styles_post_id() {
```

hellofromtonya commented on src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php:
```diff
@@ -318,9 +318,9 @@ public function prepare_item_for_response( $item, $request ) {
 		if ( $theme->get_stylesheet() === wp_get_theme()->get_stylesheet() ) {
 			// This creates a record for the current theme if not existent.
-			$id = WP_Theme_JSON_Resolver::get_user_custom_post_type_id();
+			$id = WP_Theme_JSON_Resolver::get_global_styles_post_id();
```
```suggestion
			$id = WP_Theme_JSON_Resolver::get_user_global_styles_post_id();
```

hellofromtonya committed:
Rename WP_Theme_JSON_Resolver:: get_user_custom_post_type_id() to get_user_global_styles_post_id()

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -216,28 +216,28 @@ public static function get_theme_data( $deprecated = array() ) {
 	 * @return array Custom Post Type for the user's origin config.
 	 */
-	public static function get_user_data_from_custom_post_type( $theme, $should_create_cpt = false, $post_status_filter = array( 'publish' ) ) {
+	public static function get_global_styles_from_post( $theme, $create_post = false, $post_status_filter = array( 'publish' ) ) {
```
```suggestion
	public static function get_user_data_from_wp_global_styles( $theme, $create_post = false, $post_status_filter = array( 'publish' ) ) {
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -283,28 +283,28 @@ public static function get_user_data_from_custom_post_type( $theme, $should_crea
 
 		$config   = array();
-		$user_cpt = self::get_user_data_from_custom_post_type( wp_get_theme() );
+		$user_cpt = self::get_global_styles_from_post( wp_get_theme() );
```
```suggestion
		$user_cpt = self::get_user_data_from_wp_global_styles( wp_get_theme() );
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -381,12 +381,12 @@ public static function get_merged_data( $origin = 'custom' ) {
 		}
 
-		$user_cpt = self::get_user_data_from_custom_post_type( wp_get_theme(), true );
+		$user_cpt = self::get_global_styles_from_post( wp_get_theme(), true );
```
```suggestion
		$user_cpt = self::get_user_data_from_wp_global_styles( wp_get_theme(), true );
```

hellofromtonya commented on src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php:
```diff
@@ -318,9 +318,9 @@ public function prepare_item_for_response( $item, $request ) {
+			$id = WP_Theme_JSON_Resolver::get_user_global_styles_post_id();
 		} else {
-			$user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_custom_post_type( $theme );
+			$user_cpt = WP_Theme_JSON_Resolver::get_global_styles_from_post( $theme );
```
```suggestion
			$user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( $theme );
```

hellofromtonya commented on tests/phpunit/tests/theme/wpThemeJsonResolver.php:
```diff
@@ -321,25 +321,25 @@ function test_merges_child_theme_json_into_parent_theme_json() {
 		$query_count = count( $this->queries );
 		for ( $i = 0; $i < 3; $i++ ) {
-			WP_Theme_JSON_Resolver::get_user_data_from_custom_post_type( wp_get_theme() );
+			WP_Theme_JSON_Resolver::get_global_styles_from_post( wp_get_theme() );
```
```suggestion
			WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme() );
```

hellofromtonya commented on tests/phpunit/tests/theme/wpThemeJsonResolver.php:
```diff
@@ -321,25 +321,25 @@ function test_merges_child_theme_json_into_parent_theme_json() {
 	}
 
-	function test_get_user_data_from_custom_post_type_does_not_use_uncached_queries() {
+	function test_get_global_styles_from_post_does_not_use_uncached_queries() {
```
```suggestion
	function test_get_user_data_from_wp_global_styles_does_not_use_uncached_queries() {
```

hellofromtonya commented on tests/phpunit/tests/theme/wpThemeJsonResolver.php:
```diff
@@ -321,25 +321,25 @@ function test_merges_child_theme_json_into_parent_theme_json() {
+		$this->assertEquals( 1, $query_count, 'Only one SQL query should be peformed for multiple invocations of WP_Theme_JSON_Resolver::get_global_styles_from_post()' );
 
-		$user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_custom_post_type( wp_get_theme() );
+		$user_cpt = WP_Theme_JSON_Resolver::get_global_styles_from_post( wp_get_theme() );
```
```suggestion
		$user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme() );
```

hellofromtonya commented on tests/phpunit/tests/theme/wpThemeJsonResolver.php:
```diff
@@ -321,25 +321,25 @@ function test_merges_child_theme_json_into_parent_theme_json() {
 		$this->assertEmpty( $user_cpt );
 
-		$user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_custom_post_type( wp_get_theme(), true );
+		$user_cpt = WP_Theme_JSON_Resolver::get_global_styles_from_post( wp_get_theme(), true );
```
```suggestion
		$user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme(), true );
```

hellofromtonya commented on tests/phpunit/tests/theme/wpThemeJsonResolver.php:
```diff
@@ -321,25 +321,25 @@ function test_merges_child_theme_json_into_parent_theme_json() {
 		$query_count = count( $this->queries );
 		for ( $i = 0; $i < 3; $i++ ) {
-			WP_Theme_JSON_Resolver::get_user_data_from_custom_post_type( wp_get_theme() );
+			WP_Theme_JSON_Resolver::get_global_styles_from_post( wp_get_theme() );
```
```suggestion
			WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme() );
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -283,28 +283,28 @@ public static function get_user_data_from_custom_post_type( $theme, $should_crea
 		}
-		$cache_expiration = $user_cpt ? DAY_IN_SECONDS : HOUR_IN_SECONDS;
-		wp_cache_set( $cache_key, $user_cpt ? $user_cpt['ID'] : -1, '', $cache_expiration );
+		$cache_expiration = $post ? DAY_IN_SECONDS : HOUR_IN_SECONDS;
+		wp_cache_set( $cache_key, $post ? $post['ID'] : -1, '', $cache_expiration );
```
```suggestion
		$cache_expiration = $user_cpt ? DAY_IN_SECONDS : HOUR_IN_SECONDS;
		wp_cache_set( $cache_key, $user_cpt ? $user_cpt['ID'] : -1, '', $cache_expiration );
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -283,28 +283,28 @@ public static function get_user_data_from_custom_post_type( $theme, $should_crea
 				true
 			);
-			$user_cpt    = get_post( $cpt_post_id, ARRAY_A );
+			$post        = get_post( $cpt_post_id, ARRAY_A );
```
```suggestion
			$user_cpt    = get_post( $cpt_post_id, ARRAY_A );
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -262,14 +262,14 @@ public static function get_user_data_from_custom_post_type( $theme, $should_crea
 		if ( is_array( $recent_posts ) && ( count( $recent_posts ) === 1 ) ) {
-			$user_cpt = $recent_posts[0];
-		} elseif ( $should_create_cpt ) {
+			$post = $recent_posts[0];
```
```suggestion
			$user_cpt = $recent_posts[0];
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -262,14 +262,14 @@ public static function get_user_data_from_custom_post_type( $theme, $should_crea
-		if ( -1 === $post_id && ! $should_create_cpt ) {
-			return $user_cpt;
+		if ( -1 === $post_id && ! $create_post ) {
+			return $post;
```
```suggestion
			return $user_cpt;
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -216,28 +216,28 @@ public static function get_theme_data( $deprecated = array() ) {
 			$theme = wp_get_theme();
 		}
-		$user_cpt         = array();
+		$post             = array();
```
```suggestion
		$user_cpt         = array();
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -283,28 +283,28 @@ public static function get_user_data_from_custom_post_type( $theme, $should_crea
+		wp_cache_set( $cache_key, $post ? $post['ID'] : -1, '', $cache_expiration );
 
-		return $user_cpt;
+		return $post;
```
```suggestion
		return $user_cpt;
```

hellofromtonya committed:
Revert variable renaming.

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -296,15 +296,15 @@ public static function get_user_data_from_custom_post_type( $theme, $should_crea
+	 * @return WP_Theme_JSON Entity that holds styles for the current theme.
 	 */
-	public static function get_user_data() {
+	public static function get_global_styles() {
```
```suggestion
	public static function get_user_data() {
```

oandregal replied:
Note that there's already a `get_user_data` so this needs to be different, hence, why I suggested `get_user_data_{something}`

oandregal replied:
Oh, dismiss my comment. Too many changes, I got confused :) This is just renaming it back to what it was.

hellofromtonya replied:
No worries @oandregal. Yes, it reverted it back to the original 👍

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -296,15 +296,15 @@ public static function get_user_data_from_custom_post_type( $theme, $should_crea
 	 * @since 5.9.0
 	 *
-	 * @return WP_Theme_JSON Entity that holds user data.
+	 * @return WP_Theme_JSON Entity that holds styles for the current theme.
```
```suggestion
	 * @return WP_Theme_JSON Entity that holds styles for user data.
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -339,7 +339,7 @@ public static function get_user_data() {
 	 *
 	 * Unlike the getters {@link get_core_data},
-	 * {@link get_theme_data}, and {@link get_user_data},
+	 * {@link get_theme_data}, and {@link get_global_styles},
```
```suggestion
	 * {@link get_theme_data}, and {@link get_user_data},
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -367,7 +367,7 @@ public static function get_merged_data( $origin = 'custom' ) {
 
 		if ( 'custom' === $origin ) {
-			$result->merge( self::get_user_data() );
+			$result->merge( self::get_global_styles() );
```
```suggestion
			$result->merge( self::get_user_data() );
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -216,24 +216,24 @@ public static function get_theme_data( $deprecated = array() ) {
 
 	/**
-	 * Returns the custom post type that contains the user's origin config
+	 * Returns an array that contains global styles
```
```suggestion
	 * Returns the custom post type that contains the user's origin config
```

hellofromtonya commented on src/wp-includes/class-wp-theme-json-resolver.php:
```diff
@@ -216,24 +216,24 @@ public static function get_theme_data( $deprecated = array() ) {
 	 * for the current theme or a void array if none are found.
 	 *
-	 * This can also create and return a new draft custom post type.
+	 * This can also create and return a new draft post.
```
```suggestion
	 * This can also create and return a new draft custom post type.
```

hellofromtonya committed:
Restore ::get_user_data() method name and "user" related comments.

hellofromtonya committed:
Rename ::get_user_data_from_custom_post_type() to ::get_user_data_from_wp_global_styles()

Replaces the "custom_post_type" with the actual post type of "wp_global_styles".

hellofromtonya reviewed (approved):
Marking for commit.

hellofromtonya commented:
@anton-vlasenko are you available to resolve the merge conflicts?

Commit:
Merge branch 'trunk' into fix/naming-of-methods-in-wp-theme-json-resolver-is-not-optimal

hellofromtonya commented:
Committed via changeset https://core.trac.wordpress.org/changeset/52372.
